Allow the suite parser to skip adding resources

ctsims opened the pull request:
X-Requested (but can be pulled on its own) with CommCare Android

wpride commented on backend/src/org/commcare/xml/SuiteParser.java:
```diff
@@ -75,19 +77,25 @@ public Suite parse() throws InvalidStructureException, IOException, XmlPullParse
                     } else if (parser.getName().toLowerCase().equals("xform")) {
                         //skip xform stuff for now
                         parser.nextTag();
                         Resource r = new ResourceParser(parser, maximumResourceAuthority).parse();
```
Can those line go inside the if() block?

ctsims replied:
We're stream processing, so we can't in this case, we need to consume the full document in order to move on, and can't skip stuff.

wpride commented on backend/src/org/commcare/xml/SuiteParser.java:
```diff
@@ -75,19 +77,25 @@ public Suite parse() throws InvalidStructureException, IOException, XmlPullParse
                         String path = parser.getAttributeValue(null, "path");
                         //Can be an arbitrary number of resources inside of a media block.
                         while (this.nextTagInBlock("media")) {
                             Resource r = new ResourceParser(parser, maximumResourceAuthority).parse();
```
and this line

phillipm commented on backend/src/org/commcare/xml/SuiteParser.java:
```diff
@@ -140,4 +148,12 @@ protected boolean inValidationMode() {
+    /**
+     * If set to true, the parser won't process adding incoming resources to the resource table.
+     * This is helpful if the suite is being processed during a non-install phase
+     * @param skipResources
```
Remove unused javadoc param entry

phillipm commented on backend/src/org/commcare/xml/SuiteParser.java:
```diff
@@ -33,6 +33,8 @@
     ResourceTable table;
     String resourceGuid;
 
+    private boolean skipResources = false;
```
Could fixture processing also be skipped on app init?

ctsims replied:
Hm, plausible. The app fixtures are tiny right now, so I'm inclined to keep doing it until I have a chance to consider it more deeply

phillipm commented on backend/src/org/commcare/xml/SuiteParser.java:
```diff
@@ -33,6 +33,8 @@
 
+    private boolean skipResources = false;
+
     public SuiteParser(InputStream suiteStream, ResourceTable table, String resourceGuid) throws IOException {
```
Will you add a class level doc? Something saying that this is used to used to build menus, entry actions, display details, get the app version, and install resources. I found myself wondering why we even run the suite parser on app init, and having such documentation would have answered it for me.

ctsims replied:
Sure, will add

ctsims commented:
@phillipm Addressed your comments. @wpride Unfortunately we can't quite just skip due to the way we're streaming rather than processing the full DOM. We might be able to skip the elements, but I'm not sure it's actually much faster.
